test/metabase/api/advanced_computation_test.clj

```diff
@@ -186,3 +199,208 @@
               (is (= ["CO" "Affiliate" "Gadget" 0 62 211] (nth rows 100)))
               (is (= ["ND" nil nil 6 589 2183] (nth rows 2250)))
               (is (= [nil nil nil 7 18760 69540] (last rows))))))))))
+
+(defmacro ^:private with-temp-public-dashboard {:style/indent 1} [[binding & [dashboard]] & body]
+  `(let [dashboard-settings# (merge
+                              {:parameters [{:id      "_STATE_"
+                                             :name    "State"
+                                             :slug    "state"
+                                             :type    "string"
+                                             :target  [:dimension [:fk-> (mt/id :orders :user_id) (mt/id :people :state)]]
+                                             :default nil}]}
+                              (shared-obj)
+                              ~dashboard)]
+     (mt/with-temp Dashboard [dashboard# dashboard-settings#]
+       (let [~binding (assoc dashboard# :public_uuid (:public_uuid dashboard-settings#))]
+         ~@body))))
```

Comment:
It looks like the only difference between this and the version in `metabase.api.public-test` is `:parameters`, and `with-temp-public-dashboard` lets you override stuff. So you could override `:parameters` that way. You would just need to tweak the macro so it doesn't try to eval the default parameters (because of the `(mt/id :venues :id)` form if you override it.

IMO It's better to tweak the existing test utils and make them a little more flexible so they work for your use case rather than copy the code and make a few minor changes. This way instead of 10 test utils that work in 1 place each we end up with 1 test util that works in 10 places

Reply:
something like this

```clj
(defmacro with-temp-public-dashboard {:style/indent 1} [[binding & [dashboard]] & body]
  `(let [dashboard-settings# (merge
                              ~(when-not (:parameters dashboard)
                                 `{:parameters [{:id      "_VENUE_ID_"
                                                 :name    "Venue ID"
                                                 :slug    "venue_id"
                                                 :type    "id"
                                                 :target  [:dimension (mt/id :venues :id)]
                                                 :default nil}]})
                              (shared-obj)
                              ~dashboard)]
     (mt/with-temp Dashboard [dashboard# dashboard-settings#]
       (let [~binding (assoc dashboard# :public_uuid (:public_uuid dashboard-settings#))]
         ~@body))))
```

would let you reuse it